datajoint/expression.py

```diff
@@ -639,6 +640,7 @@ def __repr__(self):
         return (
             super().__repr__()
             if config["loglevel"].lower() == "debug"
+            or getattr(self, "database", None) is None
```

Comment:
perhaps this is not the right place for this check. Derived query expressions do not have `.database`.  Only base expressions do. Perhaps you can move this logic into `Table` or `UserTable`.

Reply:
The fix above will turn off display for many derived expressions such as joins for example since the Expression object resulting from a join will not have a `table` property.

Reply:
What if instead of `.database` we check if `.heading` exists? `.heading` will always be `None` if a table is not yet activated, correct? `preview` will always require a `Heading` class.
 
https://github.com/datajoint/datajoint-python/blob/07c5553e403a3fdf13a0675a81b27ab128068aa4/datajoint/preview.py#L7

Reply:
@dimitri-yatsenko If the error is intentional (e.g., trying to instantiate a non-activated `UserTable`), maybe there should be a specific error thrown in `preview` that describes the activation issue instead of errors like: `AttributeError: 'NoneType' object has no attribute 'non_blobs'`

Reply:
I agree we need to properly handle this case. Raising an informative error sounds reasonable. We just need to test this behavior for different types of queries, including derived.